ark/internal/genai/query_parameters_test.go

```diff
@@ -0,0 +1,365 @@
+
+		query := arkv1alpha1.Query{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-query",
+				Namespace: "test-ns",
+			},
+			Spec: arkv1alpha1.QuerySpec{
+				Type: "user",
+				Parameters: []arkv1alpha1.Parameter{
+					{
+						Name: "missing_param",
+						ValueFrom: &arkv1alpha1.ValueFromSource{
+							ConfigMapKeyRef: &corev1.ConfigMapKeySelector{
+								LocalObjectReference: corev1.LocalObjectReference{
+									Name: "nonexistent-config",
+								},
+								Key: "missing-key",
+							},
+						},
+					},
+				},
+			},
+		}
+
+		// Set the input using the RawExtension helper
+		err := query.Spec.SetInputString("Hello {{.missing_param}}")
+		require.NoError(t, err)
+
+		_, err = GetQueryInputMessages(ctx, query, k8sClient)
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "failed to resolve query input")
+	})
+
+	t.Run("messages type with empty messages array", func(t *testing.T) {
+		k8sClient := fake.NewClientBuilder().WithScheme(scheme).Build()
+
+		query := arkv1alpha1.Query{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-query",
+				Namespace: "test-ns",
+			},
+			Spec: arkv1alpha1.QuerySpec{
+				Type: "messages",
+			},
+		}
+
+		// Set empty messages array using the RawExtension helper
+		err := query.Spec.SetInputMessages([]arkv1alpha1.Message{})
+		require.NoError(t, err)
+
+		messages, err := GetQueryInputMessages(ctx, query, k8sClient)
+		require.NoError(t, err)
+		require.Len(t, messages, 0)
+	})
+}
+
+func BenchmarkGetQueryInputMessages(b *testing.B) {
+	ctx := context.Background()
+	scheme := runtime.NewScheme()
+	require.NoError(b, corev1.AddToScheme(scheme))
+	require.NoError(b, arkv1alpha1.AddToScheme(scheme))
+	k8sClient := fake.NewClientBuilder().WithScheme(scheme).Build()
+
+	// Test with user type
+	userQuery := arkv1alpha1.Query{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "bench-query-user",
+			Namespace: "test-ns",
+		},
+		Spec: arkv1alpha1.QuerySpec{
+			Type: "user",
+		},
+	}
+
+	// Set input for user query using RawExtension helper
+	err := userQuery.Spec.SetInputString("Hello, this is a benchmark test message")
+	require.NoError(b, err)
+
+	// Test with messages type
+	messagesQuery := arkv1alpha1.Query{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "bench-query-messages",
+			Namespace: "test-ns",
+		},
+		Spec: arkv1alpha1.QuerySpec{
+			Type: "messages",
+		},
+	}
+
+	// Set input for messages query using RawExtension helper
+	benchMessages := []arkv1alpha1.Message{
+		{Role: "user", Content: "Hello!"},
+		{Role: "assistant", Content: "Hi there! How can I help you?"},
+		{Role: "user", Content: "This is a benchmark test"},
+	}
+	err = messagesQuery.Spec.SetInputMessages(benchMessages)
+	require.NoError(b, err)
+
+	b.Run("user_type", func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			_, err := GetQueryInputMessages(ctx, userQuery, k8sClient)
+			if err != nil {
+				b.Fatal(err)
+			}
+		}
+	})
+
+	b.Run("messages_type", func(b *testing.B) {
```

Comment:
not quite relevant for here but i think a test to make sure additional `messages` params are preserved will be useful